lib/decidim/cdtb/spam/user_spam_detector.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require "csv"
+
+module Decidim
+  module Cdtb
+    module Spam
+      # Detect spam behavior in users
+      #
+      class UserSpamDetector < ::Decidim::Cdtb::Task
+        DEFAULT_SPAM_WORDS = %w[viagra sex game free crypto crack xxx luck girls vip download].freeze
+
+        def initialize(host = nil)
+          @host = host
+          progress_bar = { title: "Decidim::User" }
+          super("SPAM DETECTOR", progress_bar: progress_bar)
+        end
+
+        def prepare_execution(_ctx)
+          if @host.present?
+            @query = Decidim::Organization.where("host ilike ?", "%#{@host}%")
+            @users = Decidim::User.where(organization: @query.first)
+          else
+            @users = Decidim::User.all
+          end
+
+          @num_users = @users.count
+          log_task_info("Checking #{@num_users} users...")
+        end
+
+        def total_items
+          @num_users
+        end
+
+        def do_execution(context)
+          progress_bar = context[:progress_bar]
+          @suspicious_users = []
+
+          @users.find_each do |user|
+            if spam_user?(user)
+              @suspicious_users << user
```

Comment:
There are organizations with thousands of spam users. The system resources may collapse if we add so many User objects into memory.

The CSV must be generated on the fly during the find_each :pray: